src/ast/passes/codegen_llvm.h

```diff
@@ -259,7 +258,11 @@ class CodegenLLVM : public Visitor {
   int current_usdt_location_index_{ 0 };
   bool inside_subprog_ = false;
 
-  std::map<std::string, AllocaInst *> variables_;
+  struct VariableLLVM {
+    llvm::Value *value;
+    llvm::Type *type;
+  };
+  std::map<std::string, VariableLLVM> variables_;
```

Comment:
Since this is class level is there a case of possible collision where two (or more probes) use the same scratch variable name and reference them in a loop?

Reply:
`variables_` is cleared between probes so they won't collide: https://github.com/bpftrace/bpftrace/blob/ca1b216ba92d007c1aa4aa1e46aa6270c8caadaf/src/ast/passes/codegen_llvm.cpp#L2547